organization/_posts/2018/2018-09-19-packaging.md

```diff
@@ -0,0 +1,88 @@
+- Ben C: How difficult was it to reuse the LCG CVMFS stack? O.k., but some issues due to underlying differences between LCG/Spack ways of doing things.
+- Graeme: Would be great to see spack used to build the LCG set of packages as the base of a chain. Javier: Can share recipes/scripts.
+
+[Dockter Spacklove - Hadrien](https://indico.cern.ch/event/754985/contributions/3144307/attachments/1718025/2772358/Spackaging.pdf)
+===========================
+- Basis for work: long term issue of building/portability in HEP, toolchain improvements to help this
+- Portablitity from CERN/WCLG p.o.v: Portable if SLC6, have CVMFS
+- Rest of world p.o.v: Portable if runs on my computer. Many Linuces/Unices, offline working
+- User profiles
+  - Testers: quick setup, o.k. With lack of host OS integration
+  - Developers: use “full” system (dev/GUI/drivers), o.k. With large builds if infrequent and automated.
+  - Both: can assume some computing proficiency
+- For testers: Containers best solution? Singularity good for scientific use, but Docker better for usability (at present)
+  - Graeme: may have Docker for development, Singularity for runtime. Hadrien: have seen some issues with running Singularity containers
+- Developers: Custom build on base system, avoid too much isolation.
+  - Playoff between integration/reproducibility.
+  - Main dependencies plus opt-in use of system packages
+- Package managers: various forms of flexibility needed for these use cases
+  - Languages, portability, environment, opt-in use of system.
+  - "Cousin" builds: Same package versions apart from build options.
+  - Spack seems most appropriate here
+- PMs + Containers:
+  - Big dockerfile -> few pm commands
+  - Testers -> devs easily via set of PM commands (vs different distros)
+- Work so far: Simplification via moving build recipes -> Dockerfiles. Trying to move most shell scripting to Spack commands.
+  - Testing by installing Verrou, Templight, ACTS, Gaudi
+- Example of Dockerfile for Gaudi: huge reduction in SLOC: 234 -> 234 (⅔ comments)
+- Conclusion: Containers + PMs help the Tester/Developer use cases and ease transition back and forth between them.
+- Javier: Will update Spack ROOT recipe ASAP.
+  - Patrick: [FNAL also have a ROOT recipe](https://github.com/FNALssi/spack/blob/fnal-develop/var/spack/repos/builtin/packages/root/package.py)
+- Ben C: Agree on Containers+PMs being friends, have seen issues with trying to run modern code natively on old (in this case SLC5) grid nodes.
+- Todd: Would like to get glibc dependency in Spack at some point.
+- Hadrien: One other use case, somewhat specific, for containers is when using a rolling Linux distro. When rolling updates compiler etc, spack etc may want/need to update/rebuild everything.
+
+AOB
+===
+- Next meeting 3 October: [https://indico.cern.ch/event/758817/](https://indico.cern.ch/event/758817/)
+  - One presentation from FNAL on Spack/Spack-dev
+  - As ever, more contributions welcome, contact Graeme and Ben
+
```

Comment:
We could perhaps all add here:

- Todd: very happy with collaboration with HEP so far, would like to prioritise important HEP use cases (very good PRs received)
  - Graeme and Todd will keep in touch to better communicate and align goals and developments